### DISCUSSION TITLE
Calc md5

### DISCUSSION THREAD

**Discussion by Bunias80** [micropython/micropython#10494] [ESP32]
Hello,

I try calculating md5 from some string. hashlib doesn't have md5 ? 
AttributeError: 'module' object has no attribute 'md5'
ImportError: can't import name md5

what lib I can use?

I use esp32, MicroPython 1.19
MicroPython v1.19.1 on 2022-06-18; ESP32 module with ESP32
Flash: esp32-20220618-v1.19.1.bin

**Comment by robert-hh** [comment]
Please tell the ~~port and~~ firmware version.

**Comment by Bunias80** [comment]
I use esp32, MicroPython 1.19
MicroPython v1.19.1 on 2022-06-18; ESP32 module with ESP32
Flash: esp32-20220618-v1.19.1.bin

**Reply by robert-hh** [reply]
Indeed it is not included, not even in the daily build. Including it requires making your own firmware build after adding
`#define MICROPY_PY_UHASHLIB_MD5             (1)`
to mpconfigport.h.

**Comment by mcl-uk** [comment]
For anyone ending up here not wanting to re-build their own firmware there's a native uPy md5 implementation with hashlib.md5 compatibility at: 
https://github.com/mcl-uk/uPython_md5lib
Maybe it can help.

**Comment by jonnor** [comment]
md5 and similar hash functions would be great to distribute as native modules! Then one could just drop them in later.
